File: longestCommonPrefix.java
public class longestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        int maxn = Integer.MAX_VALUE;
        for (String str: strs) maxn = Math.min(maxn, str.length());
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < maxn; i++) {
            char c = strs[0].charAt(i);
            boolean consistent = true;
            for (int j = 1; j < strs.length; j++) {
                if (strs[j].charAt(i) != c) {
                    consistent = false;
                    break;
                }
            }
            if (!consistent) break;
            sb.append(c);
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        String[]strs = new String[] {"dog", ""};
        longestCommonPrefix lCommonPrefix = new longestCommonPrefix();
        System.out.println(lCommonPrefix.longestCommonPrefix(strs));
    }
}

File: longestCommonPrefix.js
var longestCommonPrefix = function(strs) {
    for (let i = 0; i < strs[0].length; i++) {
        let c = strs[0].charAt(i);
        for (let j = 1; j < strs.length; j++) {
            if (i === strs[j].length || strs[j].charAt(i) !== c) {
                return strs[j].substring(0, i);
            }
        }
    }
    return strs[0];
}

console.log(longestCommonPrefix(["", "d"]))